feat: Cleanup index page, fix model typos

jjmilburn opened the pull request:
**Index**

* Include reference to GH issues that ask for input on new resource
  types/models.
* Move verbose examples from intro to their own section

**Resource types**

* Add a few missing required fields from POST examples
* Modify conflicting example parameters
* Update energy consumption/generation suggested URI to not imply power flow direction related to the hosting device

(Updates are all semantic 'patch' updates, no new features or breaking changes)

Commit:
feat: Move examples, add link to GH issues

* Include reference to GH issues that ask for input on new resource
  types/models.
* Move verbose examples from intro to their own section

Commit:
feat: Move examples, add link to GH issues

* Include reference to GH issues that ask for input on new resource
  types/models.
* Move verbose examples from intro to their own section

Commit:
fix: Typos in base resource types

* Add a few missing required fields from POST examples
* Modify conflicting example parameters
* Update energy consumption/generation suggested URI to not imply power
  flow direction related to the hosting device

Commit:
Merge branch 'dev-revise-model-descriptions-and-landing-page' of github.com:angaza/nexus-channel-models into dev-revise-model-descriptions-and-landing-page

ricehornet commented on resource_types/core/energy/102-generation/energygeneration.oas.yaml:
```diff
@@ -18,7 +18,7 @@ produces:
   #- application/vnd.ocf+cbor
   - application/json
 paths:
-  "/ein":  # suggested URI
```
After thinking about this again, I realized I was still thinking about the convention wrong. Every device can just report the energy in and out. It's weird to report "this is the energy generated to me". So I'm reverting my request to change -- I think it's fine as `ein/eout`. Sorry for the confusion.

ricehornet replied:
Also the `st` enum example still isn't a complete list?

jjmilburn replied:
Highlighting that changing back means that a solar panel would *not* host this resource (at least, the URI doesn't make as much sense then). So this is implying that the resource is hosted on a device that measures its *own* input, and energy generation devices don't report via this resource.

But thats just a convention with the URI, I guess, and we can always change the URI (on a solar panel, sure, just make the URI egen if you like)

ricehornet replied:
Yes I understand that now! As long as we're consistent I'm fine -- and I'm sure we'll learn what's most useful when we get feedback from manufacturers.

ricehornet commented:
@jjmilburn sorry for flip-flopping on the `ein/out` names -- I think I like those better actually. +1 once little suggestions are addressed.

Commit:
fix: Revise URI suggestions, correct enum list

* enum list was missing entries for validation, correct it (energy
  generation)

* URI suggestions again imply that energy flow is relevant to the
  hosting device (generation is energy in, consumption is energy out),
  suggesting that the generator devices themselves don't necessarily
  host these resources (but they certainly could, the URI is just a
  suggestion...)

Commit:
feat: Add development/testing RTR values

Many resource types might be used in pre-production contexts, give a
range of values to allow for this use
